include/pmacc/traits/GetUniqueTypeId.hpp

```diff
@@ -33,41 +33,58 @@ namespace pmacc
 namespace traits
 {
 
+/** Get next available type id
+ *
+ * Warning: is not thread-safe.
+ */
+inline uint64_t getNextId( );
+
 namespace detail
 {
 
-/** create unique id
- *
- * id is not beginning with zero
+/** Global counter for type ids
+ */
+inline uint64_t & counter()
+{
+    static uint64_t value = 0;
+    return value;
+}
+
+/** Unique id for a given type
  *
- * This class is based on
- *  - http://stackoverflow.com/a/7562583
- *  - author: MSN (edited version from Sep 27th 2011 at 3:59)
- *  - license: CC-BY-SA 3.0 (http://creativecommons.org/licenses/by-sa/3.0/)
+ * @tparam T_Type type
  */
 template<typename T_Type>
```

Comment:
Note: I have studied that whole thread on SO and do not think even the original implementation was based on that one, so removed the reference.

Reply:
I agree - the referenced code follows a different approach